http.go

```diff
@@ -51,16 +52,22 @@ func (t *transport) setAuth(a Auth) {
 	t.mu.Unlock()
 }
 
-func (t *transport) newRequest(method, path string, body io.Reader) (*http.Request, error) {
-	req, err := http.NewRequest(method, t.prefix+path, body)
-	if err != nil {
-		return nil, err
-	}
+func (t *transport) addAuth(req *http.Request) *http.Request {
 	t.mu.RLock()
 	defer t.mu.RUnlock()
 	if t.auth != nil {
 		t.auth.AddAuth(req)
 	}
+	return req
+}
+
+func (t *transport) newRequest(method, path string, body io.Reader) (*http.Request, error) {
+	req, err := http.NewRequest(method, t.prefix+path, body)
+	if err != nil {
+		return nil, err
+	}
+
+	t.addAuth(req)
```

Comment:
Woud it make sense to have the context already linked to the request at this point?